optaplanner-examples/src/test/java/org/optaplanner/examples/nqueens/solver/score/NQueensConstraintProviderTest.java

```diff
@@ -0,0 +1,148 @@
+        constraintVerifier.forConstraint(constraintProvider::horizontalConflict)
+                .given(queen1, queen2)
+                .expectReward(1);
+    }
+
+    @Test
+    public void horizontalConflictWithThreeQueens() {
+        Queen queen1 = new Queen(0, row1, column1);
+        Queen queen2 = new Queen(1, row1, column2);
+        Queen queen3 = new Queen(2, row1, column3);
+        constraintVerifier.forConstraint(constraintProvider::horizontalConflict)
+                .given(queen1, queen2, queen3)
+                .expectReward(3);
+    }
+
+    @Test
+    public void noAscendingDiagonalConflictWithOneQueen() {
+        Queen queen1 = new Queen(0, row1, column1);
+        constraintVerifier.forConstraint(constraintProvider::ascendingDiagonalConflict)
+                .given(queen1)
+                .expectNoImpact();
+    }
+
+    @Test
+    public void ascendingDiagonalConflictWithTwoQueens() {
+        Queen queen1 = new Queen(0, row1, column2);
+        Queen queen2 = new Queen(1, row2, column1);
+        constraintVerifier.forConstraint(constraintProvider::ascendingDiagonalConflict)
+                .given(queen1, queen2)
+                .expectReward(1);
+    }
+
+    @Test
+    public void ascendingDiagonalConflictWithThreeQueens() {
+        Queen queen1 = new Queen(0, row1, column3);
+        Queen queen2 = new Queen(1, row2, column2);
+        Queen queen3 = new Queen(2, row3, column1);
+        constraintVerifier.forConstraint(constraintProvider::ascendingDiagonalConflict)
+                .given(queen1, queen2, queen3)
+                .expectReward(3);
+    }
+
+    @Test
+    public void noDescendingDiagonalConflictWithOneQueen() {
+        Queen queen1 = new Queen(0, row1, column1);
+        constraintVerifier.forConstraint(constraintProvider::descendingDiagonalConflict)
+                .given(queen1)
+                .expectNoImpact();
+    }
+
+    @Test
+    public void descendingDiagonalConflictWithTwoQueens() {
+        Queen queen1 = new Queen(0, row1, column1);
+        Queen queen2 = new Queen(1, row2, column2);
+        constraintVerifier.forConstraint(constraintProvider::descendingDiagonalConflict)
+                .given(queen1, queen2)
+                .expectReward(1);
+    }
+
+    @Test
+    public void descendingDiagonalConflictWithThreeQueens() {
+        Queen queen1 = new Queen(0, row1, column1);
+        Queen queen2 = new Queen(1, row2, column2);
+        Queen queen3 = new Queen(2, row3, column3);
+        constraintVerifier.forConstraint(constraintProvider::descendingDiagonalConflict)
+                .given(queen1, queen2, queen3)
+                .expectReward(3);
+    }
+
+    private NQueens readSolution(String resource) throws IOException {
+        Path tempFile = Files.createTempFile("optaplanner-test", "xml");
+        try (InputStreamReader inputStreamReader =
+                new InputStreamReader(NQueensConstraintProviderTest.class.getResourceAsStream(resource))) {
```

Comment:
Throws NPE on my machine. Can you confirm that?

Reply:
Maybe you forgot to commit `256queens_-30.xml`?

Reply:
Doesn't, tests pass both on my machine and on Jenkins.

Reply:
Fails on Jenkins.

Reply:
I could swear it didn't. Fixing, thanks for pointing out.